cv32e40s/tests/programs/custom/clic/clic.c

```diff
@@ -0,0 +1,573 @@
+//
+// Copyright 2022 Silicon Labs, Inc.
+//
+// Licensed under the Solderpad Hardware Licence, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://solderpad.org/licenses/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+//
+// CLIC CSR access tests
+//
+/////////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <assert.h>
+
+// MUST be 31 or less
+#define NUM_TESTS 9
+#define ABORT_ON_ERROR_IMMEDIATE 0
+#define SMCLIC_ID_WIDTH 5
+
+// ---------------------------------------------------------------
+// Test prototypes - should match uint32_t <name>(uint32_t index)
+// ---------------------------------------------------------------
+uint32_t mcause_mstatus_mirror_init(uint32_t index);
+uint32_t w_mcause_mpp_r_mstatus_mpp(uint32_t index);
+uint32_t w_mstatus_mpp_r_mcause_mpp(uint32_t index);
+uint32_t w_mcause_mpie_r_mstatus_mpie(uint32_t index);
+uint32_t w_mstatus_mpie_r_mcause_mpie(uint32_t index);
+uint32_t w_mie_notrap_r_zero(uint32_t index);
+uint32_t w_mip_notrap_r_zero(uint32_t index);
+uint32_t w_mtvt_rd_alignment(uint32_t index);
+uint32_t w_mtvec_rd_alignment(uint32_t index);
+
+// Helper functions
+uint32_t set_test_status(uint32_t test_no, uint32_t val_prev);
+uint32_t run_test(uint32_t (*ptr)(), uint32_t index);
+int get_result(uint32_t res);
+uint32_t max(uint32_t a, uint32_t b);
+
+const uint32_t MSTATUS_MPP_OFFSET  = 11;
+const uint32_t MSTATUS_MPIE_OFFSET = 7;
+const uint32_t MCAUSE_MPP_OFFSET   = 28;
+const uint32_t MCAUSE_MPIE_OFFSET  = 27;
+
+const uint32_t MSTATUS_MPP_MASK  = 0x3 << MSTATUS_MPP_OFFSET;
+const uint32_t MSTATUS_MPIE_MASK = 0x1 << MSTATUS_MPIE_OFFSET;
+const uint32_t MCAUSE_MPP_MASK   = 0x3 << MCAUSE_MPP_OFFSET;
+const uint32_t MCAUSE_MPIE_MASK  = 0x1 << MCAUSE_MPIE_OFFSET;
+
+// ---------------------------------------------------------------
+// Test entry point
+// ---------------------------------------------------------------
+int main(int argc, char **argv){
+
+  uint32_t (*tests[NUM_TESTS])();
```

Comment:
Since it runs it is probably fine as it is, but doesn't it need a type for the argument?